### PR TITLE
Task 05: review id

### PR DESCRIPTION
**Pull request by omcmanus1**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -28,6 +28,45 @@ describe("/api/categories", () => {
...
+      });
+  });
+  test("GET: should respond with 404 if qeuried with invalid ID", () => {
+    return request(app)
+      .get("/api/reviews/234455")
+      .expect(404)
+      .then((response) => {
+        expect(response.body.msg).toBe("Not Found");
+      });
   });
```
Comment (Beth-McCormack) [inline __tests__/app.test.js:111]: What if the endpoint is not a number? eg /api/reviews/banana

**Review by Beth-McCormack — CHANGES_REQUESTED** [review]
see comment

File: __tests__/app.test.js [hunk 2]
```diff
@@ -59,3 +64,45 @@ describe("/api/reviews", () => {
...
 });
+
+describe("/api/reviews/:id", () => {
+  test("GET: should respond with an array containing correct single review object", () => {
+    return request(app)
+      .get("/api/reviews/2")
+      .expect(200)
+      .then(({ body }) => {
+        const { review } = body;
+        expect(review).toBeInstanceOf(Array);
```
Comment (yalicee) [inline __tests__/app.test.js]: I would like body.review to respond with an review object not an array

File: models/reviews-models.js [hunk 3]
```diff
@@ -13,3 +13,20 @@ exports.selectReviews = () => {
...
+  const queryString = `
+  SELECT reviews.review_id, reviews.title, reviews.review_body,
+    reviews.designer, reviews.review_img_url, reviews.votes, 
+    reviews.category, reviews.owner, reviews.created_at
+  FROM reviews
+  WHERE reviews.review_id = $1
+  `;
+  return db
+    .query(queryString, [reviewId])
+    .then((review) => review.rows)
```
Comment (yalicee) [inline models/reviews-models.js]: reviews.rows in an array right? See my earlier comment

File: controllers/error-handling-controllers.js [hunk 4]
```diff
@@ -1,3 +1,19 @@
...
+  if (err === "Invalid ID provided") {
+    res.status(400).send({ msg: err });
+  } else next(err);
+};
+
+exports.errorHandler404 = (err, req, res, next) => {
+  if (err === "ID not found") {
+    res.status(404).send({ msg: err });
+  } else next(err);
+};
```
Comment (yalicee) [inline controllers/error-handling-controllers.js]: this is like half of a custom error handler and half of a 404. maybe you could make it more reusable?

**Review by yalicee — APPROVED** [review]
Looks really really good but some changes I would like to see. I will approve because I trust ya but please go through and make these changes before you merge :)

**Pull request merged**